### ISSUE TITLE
Support Out of Gamut RGBColor

### ISSUE THREAD

**Issue by oneam** [tompazourek/Colourful#47]
RGBColor currently enforces values to be 0..1. I would like to work with colors that are outside of the sRGB color gamut but still valid.

Is there a way to do that?

**Comment by tompazourek** [comment]
Hi Sam, you can use other RGB working spaces than sRGB, like the ones listed below (or create your custom one):

- https://github.com/tompazourek/Colourful#rgb-working-spaces

Some of the RGB working spaces have larger gamuts, but the values for each channel are still between 0 to 1. It's just that the values correspond to a different "real" color, and the RGB working spaces with wider gamuts cover more "real" colors between the 0 and 1. Hope you know what I mean.

**Comment by oneam** [comment]
That makes perfect sense, however I was just hoping to use imaginary colors for my calculations.
Specifically I have 2 scenarios:

1. Using RGBColor to test YCbCr conversion to RGB where the YCbCr uses Studio Range.
One way of dealing with this is to assume that any color outside of the studio range is imaginary.

2. Measuring a using a camera that supports BT.2020 gamut but is configured to output BT.709 values. The color chart expected value can be stored and converted to an imaginary RGB color in the BT.709 gamut and clipped later.

**Comment by KelSolaar** [comment]
@oneam : Just a *pedantic* terminology note: Colours outside the [0, 1] range are not necessarily labeled as _imaginary_. _Imaginary_ colour is the term commonly used for colours outside the spectral locus, and thus representing invisible electromagnetic radiation that is not light!

As an example, say you captured and encoded a High Dynamic Range Image (HDRI) of some very vivid flowers directly to BT.2020 gamut. Assuming that everything fits inside the BT.2020 gamut and you did not clip your values to range [0, 1], it would be perfectly normal to have values going over 1, and preferred because we are talking about HDRI here. You need a great dynamic range in that context. Now let's say that you convert your BT.2020 to a smaller gamut such as sRGB, some of the colour cannot be represented in R>0 thus they will be negatively encoded! That does not make them _imaginary_, just not representable by the sRGB gamut.

**Comment by oneam** [comment]
Agreed, my mistake. That would be another good use of values outside of 0..1

I believe BT.2020 and BT.2100 use [1.0, 1.0, 1.0] to mean white at 100 nits and allow values higher than 1.0

**Comment by oneam** [comment]
Looks like I’m only partially right again. BT.2020 defines a range of 0..1 for most values. They still define studio range and allow signal values above and below black and white levels (super-white and sub-black) which makes using RGBColor difficult. 

There is an alternative coding scheme in BT.2100 that uses linear RGB floats to represent colours, where 1.0 represents either 1.0 cd/m2 or “the maximum diffuse white level” but that is not relevant here.
